File: project-euler/ruby/9.rb
#!/usr/bin/env ruby

def tripsum(n = 1000)
	(1..n/2).each do |i|
		(i..n-i).each do |j|
			k = n - i - j
			return [i, j, k] if i**2 + j**2 == k**2
		end
	end
end

i, j, k = tripsum
puts "#{i}^2 + #{j}^2 = #{k}^2"
puts "#{i}*#{j}*#{k} = #{i*j*k}"
#200^2 + 375^2 = 425^2
#200*375*425 = 31875000
# 31875000

File: project-euler/python/9.py
#!/usr/bin/env python

# Product of only pythagorean triplet with sum 1000

def triplet_with_sum(N):
	for i in range(N/3, N):
		for j in range(i/2, i):
			k = N - i - j
			if i*i + j*j == k*k:
				return (i, j, k)

if __name__ == '__main__':
	N = 1000
	a, b, c = triplet_with_sum(N)
	print "{0}^2 + {1}^2 = {2}^2; product = {3}".format(a, b, c, a*b*c)

